### PR TITLE
updated Delete FFY modal

### LINKED ISSUE #2996: [Dev] Update the delete process for FFY
Feature -  
Update FFY delete process.  Many delete modals have been updated, but this one has not been aligned yet.  See #2926 for background details.

### This task is done when…
This could include:
 - [ ] deleting a Fiscal year from an APD shows the desired delete modal
 
*Add a LABEL (design, dev, compliance, BUG, size, etc) before submitting.

*If the issue is needed to complete prioritized work for the CURRENT SPRINT, add it to the "This Sprint" pipeline. Otherwise, all other issues will be automatically added to the _unprioritized pipeline_ for PRODUCT to prioritize.

### PR DESCRIPTION
**Pull request by thetif**
resolves #2996 

**Description-**
Changed the FFY delete alert to match the rest of the delete modals.

**This pull request changes...**

- changed the text for the Delete FFY modal to match the mock up
- changed ApdSummary to use the modal instead of the window confirm
- updated ApdSummary tests to use RTL instead of enzyme
- added userEvent module to make tests easier to write

**Steps to manually verify this change...**

1. Login
2. Open APD

Test adding
- Click 2023
- Expected: The 2023 FFY should be checked

Test removing and canceling 
- Click 2020
- Expected: The delete modal to pop up and match the modal
- Click Cancel
- Expected: The delete modal should close and 2020 should still be checked

Test removing
- Click 2021
- Expected: The delete modal to pop up and match the modal
- Click Delte
- Expected: The delete modal should close and 2021 should not be checked

### This pull request is ready to review when...

- [X] Automated tests are updated (and all tests are passing)
- [X] Changelog is updated as appropriate
- [X] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [x] Code has been reviewed by someone other than the original author
- [x] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by knollfear — APPROVED** [review]
looks good and works as expected.

File: web/src/containers/ApdSummary.test.js [hunk 1]
```diff
@@ -1,111 +1,166 @@
-import { shallow } from 'enzyme';
 import React from 'react';
+import { renderWithConnection, screen } from '../shared/apd-testing-library';
```
Comment (tbolt) [inline web/src/containers/ApdSummary.test.js:2]: Nice switch.

Meta note: It's interesting how libraries have different philosophies on testing. For example in the "add a year test," the enzyme test focuses on a method being called. Whereas in the react-testing-library one we test the UI only. Which is probably _more_ appropriate for react.

**Review by tbolt — APPROVED** [review]

**Comment by beparticular** [comment]
@tbolt I'm having trouble logging in to this test environment. can you check on it? thx!

**Review by beparticular — APPROVED** [review]

**Review by jeromeleecms — APPROVED** [review]
This looks good. 

Note, I noticed that the "Cancel" button is now a link not a button (as shown in the mockups). We will need the update design files at some point, but no hurry as long as we are consistent - Jenn has already documented this and will add it to the work list when necessary.

**Pull request merged**